test/e2e/resize_nodes.go

```diff
@@ -560,6 +579,53 @@ var _ = Describe("Nodes [Disruptive]", func() {
 					}
 				}
 			})
+
+			// What happens in this test:
+			// 	Network connectivity on a node is cut off for 180 seconds to simulate temporary network partition
+			// Expect to observe:
+			// 1. Node is marked NotReady after timeout by nodecontroller (40seconds)
+			// 2. All pods on node are marked NotReady shortly after #1
+			// 3. Node and pods return to Ready after connectivivty recovers
+			It("All pods on the unreachable node should be marked as NotReady upon the node turn NotReady "+
+				"AND all pods should be mark back to Ready when the node get back to Ready before pod eviction timeout", func() {
+				By("choose a node - we will block all network traffic on this node")
+				options := api.ListOptions{}
+				nodes, err := c.Nodes().List(options)
+				Expect(err).NotTo(HaveOccurred())
+				filterNodes(nodes, func(node api.Node) bool {
+					return isNodeConditionSetAsExpected(&node, api.NodeReady, true)
+				})
+				if len(nodes.Items) <= 0 {
+					Failf("No node were found Ready: %d", len(nodes.Items))
+				}
+				node := nodes.Items[0]
+				opts := api.ListOptions{FieldSelector: fields.OneTermEqualSelector(client.PodHost, node.Name)}
+				if err = waitForMatchPodsCondition(c, opts, "Running and Ready", podReadyTimeout, podRunningReady); err != nil {
+					Failf("Pods on node %s are not ready and running within %v: %v", node.Name, podReadyTimeout, err)
+				}
+
+				By(fmt.Sprintf("Performing temporary network partition on node %s", node.Name))
+				inititateTemporaryNetworkPartition(c, &node)
+
+				By("Observing node and pod status change from Ready to NotReady after network partition")
+				if !waitForNodeToBe(c, node.Name, api.NodeReady, false, resizeNodeNotReadyTimeout) {
```

Comment:
This is racy, the machine this test is running on could be overloaded and the ifup script will come back up in 180s. Can you handle this another way? What you should do is:
* Setup a watch in your test
* Network partition node
* Wait for node to come up
* Go through the list of watch notifications returned and confirm you got a not ready notification